### ISSUE TITLE
Fahrordnungen vorladen

### ISSUE THREAD

**Issue by StephanieFrancke** [SchweizerischeBundesbahnen/DAS#90]
# Ausgangslage
Als Lokpersonal oder Beobacher möchte ich alle (relevanten) Fahrordnungen im verfügbaren Zeitraum vorladen, damit ich sie auch im Offline-Zustand aufrufen kann.

Antworten zu den Fragen aus der Fachgruppe vom 04.12.25:
- Wie werden die Fahrordnungen gespeichert?
Die Fahrordnungen werden in der selben lokalen DB auf dem Client gespeichert, wie alle anderen Fahrordnungen
- Was passiert, wenn der Cache voll ist? Gibt es eine Meldung?
Das würde bedeuten, dass das Tablet kein Speicherplatz mehr hat. Dies sollte von iOS und Android gemeldet werden. Im Client ist dazu nichts eingeplant. Aus meiner Sicht sollten aber nicht solche Datenmengen entstehen, dass dies zu einem Problem führt. Wenn wir ein Mengengerüst haben von TMS-VAD bezüglich SP, JP etc. müssen wir das mal durchrechnen.
- Gibt es eine manuelle Möglichkeit, den Speicher freizugeben?
Grundsätzlich können alle lokalen App-Daten über das System gelöscht werden. Innerhalb vom Client ist dazu nichts geplant bisher. Meines Wissens ist ein Clean-Up Job eingeplant, der nicht benötigte Fahrordnungen wieder löscht.

# Weiterführende Links
--
# Out of Scope
- depends on #535 
- relates to #524 
- Öffnen der Fahrordnung im Online-Zustand: #92 
- als Grundlage wird eine Liste aller Zugnummer (mit Datum und EVU) benötigt: #217 
- Fahrordnungen werden nur mit einem initialen Stand geladen, es erfolgen keine weiteren Aktualisierungen
# Akzeptanzkriterien
- Solange der DAS-Client geöffnet ist und Verbindung zum DAS-Backend besteht, werden kontinuierlich alle verfügbaren Fahrordnungen auf dem Gerät vorgeladen.
  - Es wird einmalig der Stand 4h vorher vorgeladen.
- Die vorgeladenen Fahrordnungen können in einem Offline-Zustand immer noch geöffnet werden.
# Testszenarien
## Vorbedingung:
- DAS wurde geöffnet und das Vorladen durchgeführt.
## Durchführung:
- Das Tablet wird in den Flugmodus gesetzt.
- Eine vorgeladene Fahrordnung wird geöffnet.
- Der Flugmodus wird entfernt und die Verbindung wieder aufgebaut.
  - Zur geöffneten Fahrordnung werden Updates empfangen.

# Team Review
- [x] BA Review (@StephanieFrancke)
- [x] DEV Review (@rawi-coding)
- [x] PO Review (@marcdettwiler)

**Comment by rawi-coding** [comment]
@StephanieFrancke Woher kommt dieser Anforderung?

> Die vorgeladenen Fahrordnungen sind in einem initialen Zustand und werden nicht weiter aktualisiert.
Eine Aktualisierung erfolgt jedoch, sobald die Fahrordnung geöffnet wird und das Gerät online ist.

Steht das nicht im Konflikt mit dieser Anforderung?

> Solange der DAS-Client geöffnet ist und Verbindung zum DAS-Backend besteht, werden kontinuierlich alle verfügbaren Fahrordnungen auf dem Gerät vorgeladen.

Weil wenn wir z.B. alle 5min prüfen, ob neue/geänderte Fahrordnungen geladen werden können, wäre meine Erwartung, dass wir Änderungen nicht ignorieren.

**Comment by rawi-coding** [comment]
Ja, meines Wissens holt das Backend das über den gleichen Kanal (bitte korrigieren wenn falsch @mghilardelli). Das AK würde ich entfernen 👍

**Comment by stbruni** [comment]
Hallo @StephanieFrancke 
Habe ich das richtig verstanden, dass ALLE Züge vier Stunden in der Zukunft vorgeladen werden? 
Die Frage in der Fachgruppe vom letzten Donnerstag betreffend wie lange wie die Daten noch verfügbar haben sollten bezog sich in dem Fall auch auf diesen Fall?

**Comment by StephanieFrancke** [comment]
BA Testing done, soweit beurteilbar ist alles tiptop.

**Comment by marcdettwiler** [comment]
PO Testing done 👍
